rest/config_new.go

```diff
@@ -0,0 +1,240 @@
+	Facebook *FacebookConfig `json:"facebook,omitempty"`
+	Google   *GoogleConfig   `json:"google,omitempty"`
+}
+
+type FacebookConfig struct {
+	Register bool `json:"register,omitempty" help:"If true, server will register new user accounts"`
+}
+
+type GoogleConfig struct {
+	Register    bool     `json:"register,omitempty" help:"If true, server will register new user accounts"`
+	AppClientID []string `json:"app_client_id,omitempty" help:"List of enabled client ids"`
+}
+
+type ReplicatorConfig struct {
+	MaxHeartbeat    time.Duration `json:"max_heartbeat,omitempty"    help:"Max heartbeat value for _changes request"`
+	BLIPCompression *int          `json:"blip_compression,omitempty" help:"BLIP data compression level (0-9)"`
+}
+
+type UnsupportedConfig struct {
+	StatsLogFrequency *time.Duration `json:"stats_log_frequency,omitempty" help:"How often should stats be written to stats logs"`
+	UseStdlibJSON     bool           `json:"use_stdlib_json,omitempty"     help:"Bypass the jsoniter package and use Go's stdlib instead"`
+
+	HTTP2 *HTTP2Config `json:"http2,omitempty"`
+}
+
+type HTTP2Config struct {
+	Enabled *bool `json:"enabled,omitempty" help:"Whether HTTP2 support is enabled"`
+}
+
+func LoadStartupConfigFromPath(path string) (*StartupConfig, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = f.Close() }()
+
+	var sc StartupConfig
+	err = decodeAndSanitiseConfig(f, &sc)
+	return &sc, err
+}
+
+func LoadStartupConfigFromPaths(paths ...string) (*StartupConfig, error) {
+	if len(paths) == 0 {
+		return nil, nil
+	}
+
+	sc := StartupConfig{}
+	for _, path := range paths {
+		pathSc, err := LoadStartupConfigFromPath(path)
+		if err != nil {
+			return nil, err
+		}
+		err = mergo.Merge(&sc, pathSc, mergo.WithOverride)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &sc, nil
+}
+
+// setGlobalConfig will set global variables and other settings based on the given StartupConfig.
+// We should try to keep these minimal where possible, and favour ServerContext-scoped values.
+func setGlobalConfig(sc *StartupConfig) error {
+
+	// Per-process limits, can't be scoped any narrower.
+	if os.Getenv("GOMAXPROCS") == "" && runtime.GOMAXPROCS(0) == 1 {
+		// TODO: As of Go 1.5, the runtime automatically increases GOMAXPROCS to match the number of CPUs, all of this seems unnecessary.
+		cpus := runtime.NumCPU()
+		if cpus > 1 {
+			runtime.GOMAXPROCS(cpus)
+			base.Infof(base.KeyAll, "Configured Go to use all %d CPUs; setenv GOMAXPROCS to override this", cpus)
+		}
+	}
+	if _, err := base.SetMaxFileDescriptors(sc.MaxFileDescriptors); err != nil {
+		base.Errorf("Error setting MaxFileDescriptors to %d: %v", sc.MaxFileDescriptors, err)
+	}
+
+	// Given unscoped usage of base.JSON functions, this can't be scoped.
+	if sc.Unsupported.UseStdlibJSON {
+		base.Infof(base.KeyAll, "Using the stdlib JSON package")
+		base.UseStdlibJSON = true
+	}
+
+	// TODO: Move to be scoped under auth.Authenticator ?
```

Comment:
Agree with this todo - makes sense there, and that's where users will be looking for it.